kola/tests/misc/nvidia.go

```diff
@@ -3,50 +3,209 @@ package misc
+		return false
+	}
+	return true
+}
 
+func waitForNvidiaDriver(c *cluster.TestCluster, m *platform.Machine) error {
 	nvidiaStatusRetry := func() error {
-		out, err := c.SSH(m, "systemctl status nvidia.service")
+		out, err := c.SSH(*m, "systemctl status nvidia.service")
 		if !bytes.Contains(out, []byte("active (exited)")) {
 			return fmt.Errorf("nvidia.service: %q: %v", out, err)
 		}
 		return nil
 	}
 
 	if err := util.Retry(40, 15*time.Second, nvidiaStatusRetry); err != nil {
+		return err
+	}
+	return nil
+}
+
+func verifyNvidiaInstallation(c cluster.TestCluster) {
+	m := c.Machines()[0]
+	if err := waitForNvidiaDriver(&c, &m); err != nil {
 		c.Fatal(err)
 	}
-	c.AssertCmdOutputContains(m, "/opt/bin/nvidia-smi", "Tesla")
+	out := c.MustSSH(m, "/opt/bin/nvidia-smi")
+	c.Logf("nvidia-smi: %s", out)
+}
+
+func verifyNvidiaGpuOperator(c cluster.TestCluster) {
+	params := map[string]string{}
+	// For amd64 the suffix is x86-64, for arm64 it's arm64
+	if kola.QEMUOptions.Board == "arm64-usr" {
+		params["ARCH_SUFFIX"] = "arm64"
+	} else {
+		params["ARCH_SUFFIX"] = "x86-64"
+	}
+
+	butane, err := testsutil.ExecTemplate(nvidiaOperatorTemplate, params)
+	if err != nil {
+		c.Fatalf("ExecTemplate: %s", err)
+	}
+
+	m, err := testsutil.NewMachineWithLargeDisk(c, "32G", conf.Butane(butane))
+	if err != nil {
+		c.Fatalf("Cluster.NewMachine: %s", err)
+	}
+
+	if err = waitForNvidiaDriver(&c, &m); err != nil {
+		c.Fatal(err)
+	}
+	_ = c.MustSSH(m, "sudo systemctl cat nvidia.service")
+	_ = c.MustSSH(m, "sudo systemd-sysext status")
+	c.AssertCmdOutputContains(m, "sudo systemd-sysext status", "nvidia_runtime")
+	c.AssertCmdOutputContains(m, "sudo systemd-sysext status", "nvidia-driver")
+	_ = c.MustSSH(m, `curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 \
+	&& chmod 700 get_helm.sh \
+	&& HELM_INSTALL_DIR=/opt/bin PATH=$PATH:/opt/bin ./get_helm.sh`)
+	_ = c.MustSSH(m, "sudo kubeadm init --pod-network-cidr=10.244.0.0/16")
+	_ = c.MustSSH(m, `mkdir -p $HOME/.kube
+	sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+	sudo chown $(id -u):$(id -g) $HOME/.kube/config`)
+	_ = c.MustSSH(m, "kubectl apply -f https://github.com/flannel-io/flannel/releases/latest/download/kube-flannel.yml")
+	_ = c.MustSSH(m, "kubectl taint nodes --all node-role.kubernetes.io/control-plane-")
+	_ = c.MustSSH(m, "kubectl describe nodes $HOSTNAME")
+	err = util.Retry(5, 10*time.Second, func() error {
+		out, err := c.SSH(m, "kubectl get nodes")
+		if err != nil {
+			return err
+		}
+		if strings.Contains(string(out), "NotReady") {
+			return fmt.Errorf("nodes not ready: %s", string(out))
+		}
+		return nil
+	})
+	if err != nil {
+		c.Fatalf("%v", err)
+	}
+	_ = c.MustSSH(m, "/opt/bin/helm repo add nvidia https://helm.ngc.nvidia.com/nvidia  && /opt/bin/helm repo update")
+	_ = c.MustSSH(m, `/opt/bin/helm install --wait --generate-name \
+	-n gpu-operator --create-namespace \
+	--version v24.9.2 \
```

Comment:
I think this version could also be defined as a constant, next to kubernetes and nvidia runtime version constants. Thanks.

Reply:
done